frontend/components/GameVisibilityList.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+// eslint-disable-next-line no-unused-vars
+import {AppStore} from '../storage/common';
+interface GameVisibilityListProps{
+	gameVisibilityToggleMap: Map<number, GameVisibilityToggleProps>;
+}
+const GamesVisibilityList: React.FC<GameVisibilityListProps> = (props: GameVisibilityListProps) =>{
+	const {gameVisibilityToggleMap: gamesVisibility} = props;
+
+	return (
+		<>
+			{Array.from(gamesVisibility.values()).map(game=> {
+				return <GameVisibilityToggle key={game.id} name={game.name} isVisible={game.isVisible} id={game.id} />;
+			})}
+		</>
+	);
+};
+interface GameVisibilityToggleProps {
+	id: number;
+	name: string;
+	isVisible: boolean;
+}
+
+const GameVisibilityToggle: React.FC<GameVisibilityToggleProps> = (props) => <>
+	<div>
+		<input type="checkbox" checked = {props.isVisible} name={props.name} id={`spoiler #${props.id}`}/> {props.name}
+	</div>
+
+</>;
+
+const mapStateToProps = (state: AppStore): GameVisibilityListProps => {
```

Comment:
I understand the reason you have done it this way is to move as much of the logic away from the renderer. The comments below explain some of the issues with the code as it stands, but I'd like to propose two alternatives.

----

**Alternative #1:** Just pass the map of games and map of visibility to the props and accept the logic in the renderer. It's a very simple behavior that will result in much simpler code at the cost of being less "pure".

**Alternative #2:** Simplify the model you're creating to just `{game: Game, isVisible: boolean}` and do not use `Map` in this situation, because you don't need to look things up by an index, rather you just iterate through the whole list and display it sequentially.